File: app/src/main/java/com/example/coronatracker/Room/repo.java
package com.example.coronatracker.Room;

import android.app.Application;
import android.os.AsyncTask;

import androidx.lifecycle.LiveData;

import java.util.List;

public class repo {
    private final contactDao dao;
    private final LiveData<List<indiaStateModel>> listLiveData;

    public repo(Application application) {
        database base = database.getDbINSTANCE(application);
        this.dao = base.contactDao();
        this.listLiveData = dao.getOfflineData();
    }

    public LiveData<List<indiaStateModel>> getOfflineData() {
        return listLiveData;
    }

    public void insert(indiaStateModel model) {
        new insertAsyncTask(this.dao).execute(model);
    }

    private static class insertAsyncTask extends AsyncTask<indiaStateModel, Void, Void> {
        contactDao AsyncDao;

        public insertAsyncTask(contactDao asyncDao) {
            AsyncDao = asyncDao;
        }

        @Override
        protected Void doInBackground(indiaStateModel... indiaStateModels) {
            AsyncDao.insert(indiaStateModels[0]);
            return null;
        }
    }

    public void deleteAll() {
        new deleteAsync(this.dao).execute();
    }

    private class deleteAsync extends AsyncTask<Void, Void, Void> {
        private final contactDao AsyncDao;

        public deleteAsync(contactDao asyncDao) {
            AsyncDao = asyncDao;
        }

        @Override
        protected Void doInBackground(Void... voids) {
            AsyncDao.deleteAll();
            return null;
        }
    }
}

File: app/src/main/java/com/example/coronatracker/room/Repo.kt
package com.example.coronatracker.room

import android.app.Application
import androidx.lifecycle.LiveData

class Repo(application: Application?) {
    private val dao: ContactDao
    val offlineData: LiveData<List<indiaStateModel?>?>?

    suspend fun insert(model: indiaStateModel?) = dao.insert(model)

    suspend fun deleteAll() = dao.deleteAll()

    suspend fun getOfflineDataB() : List<indiaStateModel> = dao.getOfflineDataB()

    suspend fun insertState(state: StateRecent) = dao.insertState(state)

    suspend fun insertCountry(country: CountryRecent) = dao.insertCountry(country)

    fun getRecentState() : LiveData<List<StateRecent>> =dao.getRecentState()

    fun getRecentCountry(): LiveData<List<CountryRecent>> =dao.getRecentCountry()

    init {
        val base = database.getDbINSTANCE(application)
        dao = base.contactDao()
        offlineData = dao.getOfflineData()
    }
}